lib/metasploit/framework/data_service/proxy/host_data_proxy.rb

```diff
@@ -88,4 +100,25 @@ def valid(opts)
     return true
   end
 
+  def use_id_or_get_id_from_host_address(opts, method)
+    begin
+      if opts[:id]
+        self.data_service_operation do |data_service|
+          data_service.send(method, opts)
+        end
+      elsif opts[:address]
+        self.data_service_operation do |data_service|
+          host = data_service.get_host(opts)
```

Comment:
can this return nil? if it can it would skip over the next statement and not have any logging to indicate something went wrong

Reply:
So this can produce nil but that's at least expected behaviour for `get`ting a host tag. Will look into what the behaviour is for other `delete` & `add` commands

Reply:
Calling `delete` and `add` with a non-existent host address doesn't indicate to the user that no operation was carried out. If added these error messages to make an incorrect host address more apparent:
![image](https://user-images.githubusercontent.com/54621924/80390042-cf77c080-88a3-11ea-98a6-75dcc24bb6d1.png)